File: frontend/lib/models/evento_model.dart
class Evento {
  final int? id;
  final int idModalidadeEvento;
  final int idGrupoEvento;
  final int idPremiacaoEvento;
  final int idUsuario;
  final String nome;
  final String descricao;
  final String local;
  final String? capaEvento;
  final String dataInicioEvento;
  final String dataFimEvento;
  final bool isentoPagamento;
  final String dataInicioInscricoes;
  final String dataFimInscricoes;
  final double valorEvento;
  final bool situacao;
  bool isSubscribed;

  Evento({
    this.id,
    required this.idModalidadeEvento,
    required this.idGrupoEvento,
    required this.idPremiacaoEvento,
    required this.idUsuario,
    required this.nome,
    required this.descricao,
    required this.local,
    this.capaEvento,
    required this.dataInicioEvento,
    required this.dataFimEvento,
    required this.isentoPagamento,
    required this.dataInicioInscricoes,
    required this.dataFimInscricoes,
    required this.valorEvento,
    required this.situacao,
    this.isSubscribed = false,
  });

  factory Evento.fromJson(Map<String, dynamic> json) {
    return Evento(
      id: json['ID'],
      idModalidadeEvento: json['ID_MODALIDADE_EVENTO'],
      idGrupoEvento: json['ID_GRUPO_EVENTO'],
      idPremiacaoEvento: json['ID_PREMIACAO_EVENTO'],
      idUsuario: json['ID_USUARIO'],
      nome: json['NOME'],
      descricao: json['DESCRICAO'],
      local: json['LOCAL'],
      capaEvento: json['CAPA_EVENTO'],
      dataInicioEvento: json['DATA_INICIO_DESAFIO'],
      dataFimEvento: json['DATA_FIM_DESAFIO'],
      isentoPagamento: json['ISENTO_PAGAMENTO'],
      dataInicioInscricoes: json['DATA_INICIO_INSCRICAO'],
      dataFimInscricoes: json['DATA_FIM_INSCRICAO'],
      valorEvento: double.parse(json['VALOR_EVENTO']),
      situacao: json['SITUACAO'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'ID': id,
      'ID_MODALIDADE_EVENTO': idModalidadeEvento,
      'ID_GRUPO_EVENTO': idGrupoEvento,
      'ID_PREMIACAO_EVENTO': idPremiacaoEvento,
      'ID_USUARIO': idUsuario,
      'NOME': nome,
      'DESCRICAO': descricao,
      'LOCAL': local,
      'CAPA_EVENTO': capaEvento,
      'DATA_INICIO_DESAFIO': dataInicioEvento,
      'DATA_FIM_DESAFIO': dataFimEvento,
      'ISENTO_PAGAMENTO': isentoPagamento,
      'DATA_INICIO_INSCRICAO': dataInicioInscricoes,
      'DATA_FIM_INSCRICAO': dataFimInscricoes,
      'VALOR_EVENTO': valorEvento,
      'SITUACAO': situacao,
    };
  }
}

File: backend/models/eventoModel.js
const { DataTypes } = require('sequelize');
const db = require('../database/db');
const ModalidadeEvento = require('./modalidadeModel'); // Modelo de modalidades de eventos
const GrupoEvento = require('./gruposEventoModel'); // Modelo de grupos de eventos
const DuracaoEvento = require('./duracaoEventoModel'); // Modelo de duração do evento
const PremiacaoEvento = require('./premiacaoModel'); // Modelo de premiação do evento
const Usuario = require('./usuarioModel'); // Modelo de usuário

const Evento = db.define('Evento', {
    ID: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    ID_MODALIDADE_EVENTO: {
        type: DataTypes.INTEGER,
        references: {
            model: ModalidadeEvento,
            key: 'ID'
        },
        allowNull: false
    },
    ID_GRUPO_EVENTO: {
        type: DataTypes.INTEGER,
        references: {
            model: GrupoEvento,
            key: 'ID'
        },
        allowNull: false
    },
    ID_PREMIACAO_EVENTO: {
        type: DataTypes.INTEGER,
        references: {
            model: PremiacaoEvento,
            key: 'ID'
        },
        allowNull: false
    },
    ID_USUARIO: {
        type: DataTypes.INTEGER,
        references: {
            model: Usuario,
            key: 'ID'
        },
        allowNull: false
    },
    NOME: {
        type: DataTypes.STRING(100),
        allowNull: false
    },
    DESCRICAO: {
        type: DataTypes.STRING(500),
        allowNull: true
    },
    LOCAL: {
        type: DataTypes.STRING(500),
        allowNull: true
    },
    CAPA_EVENTO: {
        type: DataTypes.STRING(100), // Armazenar caminho do arquivo ou URL
        allowNull: true
    },
    DATA_INICIO_DESAFIO: {
        type: DataTypes.STRING(100),
        allowNull: true
    },
    DATA_FIM_DESAFIO: {
        type: DataTypes.STRING(100),
        allowNull: true
    },
    ISENTO_PAGAMENTO: {
        type: DataTypes.BOOLEAN,
        defaultValue: false
    },
    DATA_INICIO_INSCRICAO: {
        type: DataTypes.STRING(100),
        allowNull: true
    },
    DATA_FIM_INSCRICAO: {
        type: DataTypes.STRING(100),
        allowNull: true
    },
    VALOR_EVENTO: {
        type: DataTypes.DECIMAL(10, 2),
        allowNull: true
    },
    SITUACAO: {
        type: DataTypes.BOOLEAN,
        defaultValue: true
    },
    CRIADO_EM: {
        type: DataTypes.DATE,
        allowNull: false,
        defaultValue: DataTypes.NOW
    },
    ATUALIZADO_EM: {
        type: DataTypes.DATE,
        allowNull: true,
        defaultValue: DataTypes.NOW
    }
}, {
    tableName: 'EVENTOS',
    timestamps: true, // Sequelize automaticamente preenche os campos de data de criação e atualização
    createdAt: 'CRIADO_EM',
    updatedAt: 'ATUALIZADO_EM'
});

module.exports = Evento;
